src/components/TldrawCanvas.tsx

```diff
@@ -486,15 +474,59 @@ const TldrawCanvas = ({ title }: Props) => {
           return;
         clearTimeout(deserializeRef.current);
         deserializeRef.current = window.setTimeout(() => {
-          store.deserialize(JSON.parse(window.atob(state)));
+          store.deserialize(state);
         }, THROTTLE);
       },
     ];
-    window.roamAlphaAPI.data.addPullWatch(...props);
+    window.roamAlphaAPI.data.addPullWatch(...pullWatchProps);
     return () => {
-      window.roamAlphaAPI.data.removePullWatch(...props);
+      window.roamAlphaAPI.data.removePullWatch(...pullWatchProps);
     };
   }, [initialState, store]);
+  useEffect(() => {
+    const actionListener = ((
+      e: CustomEvent<{
+        action: string;
+        uid: string;
+        val: string;
+        onRefresh: () => void;
+      }>
+    ) => {
+      if (!/canvas/i.test(e.detail.action)) return;
```

Comment:
the way that query builder actions work is that they fire a global event, which our tldraw listens to for any `canvas` type actions